server/banned.go

```diff
@@ -37,11 +37,12 @@ func (pi *packetInfo) checkBanMessage() error {
 		}
 
 		bannedTrackerData.mutex.Lock()
-		defer bannedTrackerData.mutex.Unlock()
 		bannedTrackerData.bannedBy[td.verificationKey] = nil
+		bannedTrackerData.mutex.Unlock()
```

Comment:
If I understand correctly, there is another issue here. `bannedBy` is never cleared. Since the pool-based `banned()` test (which is used everywhere) is based on this count, this player will be effectively banned forever on this server even though the IP ban is removed by the cleanup routine.

My first thought would be to clear it in the IP-ban cleanup routine but I didn't think about it carefully yet.

Reply:
But if the connection is not long lived anyway, I guess it's a non issue.

Reply:
Correct it isn't cleared because this is cleaned up once the shuffle is completed. They will disconnect and their trackerData will get cleaned up.